src/middlewares/delete-project.middleware.js

```diff
@@ -0,0 +1,86 @@
+// @flow
+import { remote } from 'electron';
+import {
+  SHOW_DELETE_PROJECT_PROMPT,
+  DELETE_PROJECT_FROM_DISK,
+  FINISH_DELETING_PROJECT_FROM_DISK,
+  deleteProjectFromDisk,
+  deleteProjectFromDiskFinish,
+  refreshProjects,
+  selectProject,
+  createNewProjectStart,
+} from '../actions';
+
+import { getProjectsArray } from '../reducers/projects.reducer';
+
+const { dialog, shell } = remote;
+
+export default (store: any) => (next: any) => (action: any) => {
+  // eslint-disable-next-line default-case
+  switch (action.type) {
+    case SHOW_DELETE_PROJECT_PROMPT: {
+      dialog.showMessageBox(
+        {
+          type: 'warning',
+          buttons: ['Delete from Disk', 'Cancel'],
+          defaultId: 0,
+          title: `Delete ${action.project.name}`,
+          message: `Are you sure you want to delete ${action.project.name}?`,
+          detail: 'WARNING! Deleting from disk will send the project to trash!',
+        },
+        (response: number) => {
+          // TODO: Eventually need to do shouldDeleteFromGuppy as well
+          const shouldDeleteFromDisk = response === 0;
+
+          if (shouldDeleteFromDisk) {
+            const { project } = action;
+            store.dispatch(deleteProjectFromDisk(project));
```

Comment:
I think we can kill the bottom 2 cases :) this'll set up up well for the saga.

Essentially it's just a matter of squashing the 3 together, and omitting the `DELETE_PROJECT_FROM_DISK` dispatch altogether.


```js
case SHOW_DELETE_PROJECT_PROMPT:
  dialog.showMessageBox(..., response => {
    if (!shouldDeleteFromDisk) {
      return;
    }

    const successfullyDeleted = shell.moveItemToTrash(path);

    if (successfullyDeleted) {
      store.dispatch(deleteProjectFromDiskFinish(project));
    }
    // code to dispatch either selectProject or createNewProjectWizard
  })
```

I agree with @AWolf81's comment below, I don't think we need `refreshProjects`. Instead I'd update the projects reducer to remove the project.

I did this in a local copy of your branch, and the code looks like this (in `projects.reducer.js`):

```js
    case FINISH_DELETING_PROJECT_FROM_DISK: {
      const { deletedProject } = action;

      return produce(state, draftState => {
        delete draftState[deletedProject.id];
      });
    }
```

(note that it would just use `projectId` instead of `deletedProject.id` if you follow my earlier comment)